compiler/rustc_errors/src/lib.rs

```diff
@@ -892,6 +910,16 @@ impl HandlerInner {
         self.delay_as_bug(diagnostic)
     }
 
+    #[track_caller]
```

Comment:
The `#[track_caller]` on `delay_span_bug` is used only for  `std::panic::Location::caller()`, which is here replaced with a backtrace.